lightning/src/ln/channel.rs

```diff
@@ -2200,9 +2200,11 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			match err {
 				None => {
 					if update_add_htlcs.is_empty() && update_fulfill_htlcs.is_empty() && update_fail_htlcs.is_empty() && self.holding_cell_update_fee.is_none() {
-						// This should never actually happen and indicates we got some Errs back
-						// from update_fulfill_htlc/update_fail_htlc, but we handle it anyway in
-						// case there is some strange way to hit duplicate HTLC removes.
+						// Reaching this clause is an edge case: a user must (a) be at the edge
```

Comment:
Wait, hmm, shouldn't we fail this HTLC back if we fail to add it? I think we're loose an HTLC now which we will never fail/fulfill/forward and eventually get our channel closed on us by our counterparty.

Reply:
Ah, I'd agree. Gonna add this before updating the rest of the test

Reply:
@TheBlueMatt  LMK if the new logic looks good and I can add more tests.